### PR TITLE
Added header/footer.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-5284](https://openedx.atlassian.net/browse/ECOM-5284)

As a Publisher user, I would like to see a consistent edX Header/Footer as I navigate throughout the Publisher experience. 

**Design:** http://cim31f.axshare.com/course_runs.html

**Acceptance Criteria:**
1. Validate the header/footer matches the wireframe.
2. Validate the links on the header/footer work.
3. Validate the header/footer is persistent throughout every page as navigating through the Publisher.

screencapture-localhost-8004-publisher-course_runs-1475567322324

### CODE REVIEW

**Comment by waheedahmed** [comment]
@awais786 @bderusha Please review.

File: course_discovery/templates/footer.html [hunk 1]
```diff
@@ -1,6 +1,42 @@
...
+        <div class="footer-content-primary">
+          <div class="colophon">
+            {% now "Y" as current_year %}
+            <p>&copy; {{ current_year }} <a data-rel="edx.org" href="https://wwww.edx.org" rel="external">{{ platform_name }}</a>.</p>
+          </div>
+
+            <nav class="nav-peripheral">
+              <ol>
+              <li class="nav-item nav-peripheral-tos">
+                <a data-rel="edx.org" href="https://www.edx.org/edx-terms-service">{% trans "Terms of Service" %}</a>
```
Comment (bderusha) [inline course_discovery/templates/footer.html]: we can't hardcode edx links.  These need to be dynamic

File: course_discovery/templates/footer.html [hunk 2]
```diff
@@ -1,6 +1,42 @@
...
+              </li>
+              <li class="nav-item nav-peripheral-pp">
+                <a data-rel="edx.org" href="https://www.edx.org/edx-privacy-policy">{% trans "Privacy Policy" %}</a>
```
Comment (bderusha) [inline course_discovery/templates/footer.html]: same here

File: course_discovery/templates/footer.html [hunk 3]
```diff
@@ -1,6 +1,42 @@
...
+        </div>
+
+        <div class="footer-content-secondary" aria-label="{% trans 'Legal' %}">
+          <div class="footer-about-copyright">
+            <!--Site operators: Please do not remove this paragraph! This attributes back to edX and makes your acknowledgement of edX's trademarks clear.-->
+            <p>
+                <!--Translators: 'EdX', 'edX', 'Studio', and 'Open edX' are trademarks of 'edX Inc.'. Please do not translate any of these trademarks and company names.-->
+                {% with link_start='<a data-rel="edx.org" href="https://www.edx.org/">' link_end='</a>' %}
+                    {% blocktrans %}
+                        EdX, Open edX, Studio, and the edX and Open edX logos are registered trademarks or trademarks of {{link_start}}edX Inc.{{link_end}}
```
Comment (bderusha) [inline course_discovery/templates/footer.html]: What about white label?  How does this work?  @mattdrayer ?

**Comment by awais786** [comment]
👍

File: course_discovery/static/sass/publisher/course_detail.scss [hunk 4]
```diff
@@ -65,6 +202,78 @@ $light-gray: rgba(204, 204, 204, 1);
...
+                .footer-about-copyright {
+                    width: 31.91489%;
+                    float: left;
+                    margin-right: 14.28571%;
+                    display: inline-block;
+                    vertical-align: middle;
+                }
+
+                .footer-about-openedx {
+                    float: right;
```
Comment (ahsan-ul-haq) [inline course_discovery/static/sass/publisher/course_detail.scss]: use float(right)

File: course_discovery/static/sass/publisher/course_detail.scss [hunk 5]
```diff
@@ -65,6 +202,78 @@ $light-gray: rgba(204, 204, 204, 1);
...
+            .footer-content-secondary {
+                margin-top: 20px;
+                display: table;
+                width: 100%;
+                clear: both;
+
+                .footer-about-copyright {
+                    width: 31.91489%;
+                    float: left;
+                    margin-right: 14.28571%;
```
Comment (ahsan-ul-haq) [inline course_discovery/static/sass/publisher/course_detail.scss]: use `@include margin-right(14.28571%);`

File: course_discovery/static/sass/publisher/course_detail.scss [hunk 6]
```diff
@@ -65,6 +202,78 @@ $light-gray: rgba(204, 204, 204, 1);
...
+                }
+
+                .nav-peripheral {
+                    width: 48.93617%;
+                    float: right;
+                    text-align: right;
+
+                    .nav-item {
+                        display: inline-block;
+                        margin-right: 20px;
```
Comment (ahsan-ul-haq) [inline course_discovery/static/sass/publisher/course_detail.scss]: use `@include margin-right(20px);`

File: course_discovery/static/sass/publisher/course_detail.scss [hunk 7]
```diff
@@ -65,6 +202,78 @@ $light-gray: rgba(204, 204, 204, 1);
...
+
+            .footer-content-secondary {
+                margin-top: 20px;
+                display: table;
+                width: 100%;
+                clear: both;
+
+                .footer-about-copyright {
+                    width: 31.91489%;
+                    float: left;
```
Comment (ahsan-ul-haq) [inline course_discovery/static/sass/publisher/course_detail.scss]: use `@include float(left)`

File: course_discovery/templates/footer.html [hunk 8]
```diff
@@ -1,6 +1,42 @@
...
-        Copyright &copy; {{ current_year }} {{ platform_name }}.
-    </p>
-</footer>
+{% load i18n %}
+<div class="wrapper-footer wrapper">
+    <footer class="primary">
+        <div class="footer-content-primary">
+          <div class="colophon">
+            {% now "Y" as current_year %}
+            <p>&copy; {{ current_year }} <a data-rel="edx.org" href="https://wwww.edx.org" rel="external">{{ platform_name }}</a>.</p>
```
Comment (ahsan-ul-haq) [inline course_discovery/templates/footer.html]: I don't think href should be hardcoded here.

Comment (bderusha) [inline course_discovery/templates/footer.html]: Can't hardcode `www.edx.org` here.  What if the PLATFORM is not edx?

**Review by bderusha — CHANGES_REQUESTED** [review]
Code looks fine.  I have questions about the content.  How is this supposed to work for openedx? white label?

File: course_discovery/templates/footer.html (hunk 3 above)
Comment (mattdrayer) [reply course_discovery/templates/footer.html]: @douglashall should take a look as well -- in the LMS code we have a setting called PLATFORM_NAME which we use to cover certain presentation scenarios.

Comment (douglashall) [reply course_discovery/templates/footer.html]: In LMS we have a settings override mechanism which makes use of a `SiteConfiguration` model which allows us to override settings like `PLATFORM_NAME`. We would like to get to a point where there is a python package that contains multitenant related utilities like this. We are not there yet. For the time being you could implement a similar settings override mechanism in this service or you could start the multitenant python package.

File: course_discovery/static/sass/publisher/course_detail.scss (hunk 6 above)
Comment (clintonb) [reply course_discovery/static/sass/publisher/course_detail.scss]: Why the use of `@include` for basic CSS properties?

File: course_discovery/templates/footer.html (hunk 3 above)
Comment (mattdrayer) [reply course_discovery/templates/footer.html]: Does it make sense to leverage the Partner model here?  That's essentially the multi-tenant concept in this service:  https://github.com/edx/course-discovery/blob/master/course_discovery/apps/core/models.py#L59

Comment (douglashall) [reply course_discovery/templates/footer.html]: I'd prefer to stick with the Site/SiteConfiguration paradigm to solve the specific problem that is being raised here.

File: course_discovery/static/sass/publisher/course_detail.scss (hunk 6 above)
Comment (ahsan-ul-haq) [reply course_discovery/static/sass/publisher/course_detail.scss]: I think we need to provide rtl support if we don't need that now then its fine with me.

File: course_discovery/templates/footer.html (hunk 8 above)
Comment (waheedahmed) [reply course_discovery/templates/footer.html]: @bderusha I thought it is an in house app only for edX.

File: course_discovery/static/sass/publisher/course_detail.scss (hunk 6 above)
Comment (clintonb) [reply course_discovery/static/sass/publisher/course_detail.scss]: Got it. I haven't dealt with RTL, so I'll defer to you.

**Comment by waheedahmed** [comment]
@bderusha Removed the hardcoded edX links and also addressed other comments, please take another look.
Created a new story [ECOM-5959](https://openedx.atlassian.net/browse/ECOM-5959) to generate dynamic links using Site/SiteConfiguration or Partner model.

**Review by bderusha — APPROVED** [review]
Provided that we do the additional work to fix the links in https://openedx.atlassian.net/browse/ECOM-5959 this looks good to me

**Pull request merged**